ts/screens/modal/UpdateAppModal.tsx

```diff
@@ -45,130 +46,105 @@ const styles = StyleSheet.create({
-
-  public componentDidMount() {
-    BackHandler.addEventListener("hardwareBackPress", this.handleBackPress);
-  }
+const AppleFooter: FC<FooterProps> = ({ onOpenAppStore }: FooterProps) => (
+  <View footer>
+    <>
+      <Button block={true} primary={true} onPress={onOpenAppStore}>
+        <Text>{I18n.t("btnUpdateApp")}</Text>
+      </Button>
+      <View spacer />
+    </>
+  </View>
+);
 
-  public componentWillUnmount() {
-    if (this.idTimeout) {
-      clearTimeout(this.idTimeout);
-    }
-    BackHandler.removeEventListener("hardwareBackPress", this.handleBackPress);
-  }
+const AndroidFooter: FC<FooterProps> = ({ onOpenAppStore }: FooterProps) => {
+  const cancelButtonProps = {
+    cancel: true,
+    block: true,
+    onPress: () => BackHandler.exitApp(),
+    title: I18n.t("global.buttons.close")
+  };
 
-  private openAppStore = () => {
-    // the error is already displayed
-    if (this.state.hasError) {
-      return;
-    }
-    // storeUrl is not a webUrl, try to open it
-    Linking.openURL(storeUrl).catch(() => {
-      // Change state to show the error message
-      this.setState({
-        hasError: true
-      });
-      // After 5 seconds restore state
-      // eslint-disable-next-line
-      this.idTimeout = setTimeout(() => {
-        this.setState({
-          hasError: false
-        });
-      }, timeoutErrorMsg);
-    });
+  const updateButtonProps = {
+    block: true,
+    primary: true,
+    onPress: onOpenAppStore,
+    title: I18n.t("btnUpdateApp")
   };
 
-  /**
-   * Footer iOS button
-   */
-  private renderIosFooter() {
-    return (
-      <View footer={true}>
-        <React.Fragment>
-          <Button block={true} primary={true} onPress={this.openAppStore}>
-            <Text>{I18n.t("btnUpdateApp")}</Text>
-          </Button>
-          <View spacer={true} />
-        </React.Fragment>
-      </View>
-    );
-  }
+  return (
+    <FooterWithButtons
+      type="TwoButtonsInlineThird"
+      leftButton={cancelButtonProps}
+      rightButton={updateButtonProps}
+    />
+  );
+};
 
-  /**
-   * Footer Android buttons
-   */
-  private renderAndroidFooter() {
-    const cancelButtonProps = {
-      cancel: true,
-      block: true,
-      onPress: () => BackHandler.exitApp(),
-      title: I18n.t("global.buttons.close")
-    };
-    const updateButtonProps = {
-      block: true,
-      primary: true,
-      onPress: this.openAppStore,
-      title: I18n.t("btnUpdateApp")
-    };
-
-    return (
-      <FooterWithButtons
-        type="TwoButtonsInlineThird"
-        leftButton={cancelButtonProps}
-        rightButton={updateButtonProps}
-      />
-    );
-  }
+const UpdateAppModal: React.FC = () => {
+  // Disable Android back button
+  const handleBackPress = () => true;
 
-  // Different footer according to OS
-  get footer() {
-    return Platform.select({
-      ios: this.renderIosFooter(),
-      android: this.renderAndroidFooter()
-    });
-  }
+  useEffect(() => {
```

Comment:
We already have a common hook for this case `useHardwareBackButton` let's try this one 😄